tests/test_performance.py

```diff
@@ -0,0 +1,56 @@
+from math import log10
+import time
+import timeit
+
+import pytest
+
+from uncertainties import ufloat
+
+
+def repeated_summation(num):
+    """
+    generate and sum many floats together, then calculate the standard deviation of the
+    output. Under the lazy expansion algorithm, the uncertainty remains non-expanded
+    until a request is made to calculate the standard deviation.
+    """
+    result = sum(ufloat(1, 0.1) for _ in range(num)).std_dev
+    return result
+
+
+def test_repeated_summation_complexity():
```

Comment:
I've seen a test similar to this that kept randomly failing so I don't know how reliable this test will be. 
I think test_repeated_summation_speed will catch any increases in time so these tests are testing the same thing. I'm not opposed to this test- we can leave it in for now, and if it we find it unreliable we can remove it.

I'll add a link to PR with the graph you plotted as it helps makes sense of this test. It'll be good to see this plotted for your other PR too.